src/script/script_object.h

```diff
@@ -15,26 +15,65 @@ namespace script {
   struct ObjectProperty {
     Function getter;
     Function setter;
+    std::string doc;
+
+    ObjectProperty& documentation(const std::string& str) {
+      doc = str;
+      return *this;
+    }
+  };
+
+  struct DocumentedFunction : public Function {
+    std::string doc;
+
+    struct ArgDoc {
+      std::string name;
+      std::string doc;
+    };
+
+    std::vector<ArgDoc> argDoc;
+
+    std::string resultDoc = "Nothing";
+
+    DocumentedFunction(const Function& other) : Function(other) {}
+
+    DocumentedFunction(Function&& other) : Function(std::move(other)) {}
+
+    DocumentedFunction& documentation(const std::string& str) {
+      doc = str;
+      return *this;
+    }
+
+    DocumentedFunction& arg(const std::string& arg, const std::string& doc) {
```

Comment:
it took me a moment to understand it was simply a documentation function when I've seen it up there.  `docArg`?